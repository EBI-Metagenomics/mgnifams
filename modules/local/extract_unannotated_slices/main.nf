process EXTRACT_UNANNOTATED_SLICES {
    tag "$meta.id"
    label "process_single"

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/31/313e1c18a344323886cf97a151ab66d81c1a146fb129558cb9382b69a72d5532/data' :
        'community.wave.seqera.io/library/python:b1b4b1f458c605bb' }"

    input:
    tuple val(meta), path(sequence_chunk)
    val min_sequence_length

    output:
    tuple val(meta), path("${prefix}.fa"), emit: fa
    path "versions.yml"                  , emit: versions

    script:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    extract_unannotated_slices.py \\
        --input_file ${sequence_chunk} \\
        --output_file ${prefix}.fa \\
        --min_sequence_length ${min_sequence_length}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.fa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$(python --version 2>&1 | sed 's/Python //g')
    END_VERSIONS
    """
}
