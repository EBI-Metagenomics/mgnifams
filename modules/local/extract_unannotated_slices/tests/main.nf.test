nextflow_process {

    name "Test Process EXTRACT_UNANNOTATED_SLICES"
    script "../main.nf"
    process "EXTRACT_UNANNOTATED_SLICES"

    tag "modules"
    tag "modules_local"
    tag "extract_unannotated_slices"

    test("sequence_explorer_protein - csv") {

        when {
            process {
                """
                input[0] = [ [ id:'test'],
                    file("${moduleDir}/tests/sequence_explorer_protein_mgnifams_100.csv", checkIfExists: true)
                ]
                input[1] = 100
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("sequence_explorer_protein - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [ [ id:'test'],
                    file("${moduleDir}/tests/sequence_explorer_protein_mgnifams_100.csv", checkIfExists: true)
                ]
                input[1] = 100
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
