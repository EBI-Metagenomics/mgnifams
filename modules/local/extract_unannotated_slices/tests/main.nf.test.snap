{
    "sequence_explorer_protein - csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,856212330791d51c4d585d0075c422b5"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f1294cd766c5be4cdd4ff8911859573d"
                ],
                "fa": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,856212330791d51c4d585d0075c422b5"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f1294cd766c5be4cdd4ff8911859573d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T08:24:35.061440003"
    },
    "sequence_explorer_protein - csv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,f1294cd766c5be4cdd4ff8911859573d"
                ],
                "fa": [
                    [
                        {
                            "id": "test"
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f1294cd766c5be4cdd4ff8911859573d"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-20T08:24:39.151323684"
    }
}